src/lib.rs

```diff
@@ -1,21 +1,23 @@
-use crate::integrations::tex::execute;
+use crate::integrations::executable::Executable;
+use crate::integrations::sn::SnIntegration;
+use crate::integrations::tex::TexIntegration;
+use crate::models::resume::Resume;
+use clap::ArgMatches;
```

Comment:
IMHO the command-line parsing should be kept out of the lib. Think of the library as something that can be used in multiple ways. Building a command-line tool is one way to use it and that's what we do in main.rs. Someone might want to use this lib from a web service, so the lib's interface should not deal with command-line arguments.

Reply:
Agree, and yes, in general the `clap` imports should be contained in `main.rs`. The motivation was to keep `main.rs` as slim as possible and move as much the "logic" elsewhere. With a lot of tutorials cramming everything into `main.rs` have grown to dislike huge `main`s. On the other hand, didn't like the idea of `main` importing `FooIntegration`s for it's kind of `lib`s job to execute details. Will try to solve this with generics.